pkg/maintainer/bitcoin_difficulty.go

```diff
@@ -138,19 +138,6 @@ func (bdm *bitcoinDifficultyMaintainer) verifySubmissionEligibility() error {
 		return errNoGenesis
 	}
 
-	authorizationRequired, err := bdm.chain.AuthorizationRequired()
-	if err != nil {
-		return fmt.Errorf(
-			"cannot check whether authorization is required to submit "+
-				"block headers: [%w]",
-			err,
-		)
-	}
-
-	if !authorizationRequired {
-		return nil
-	}
-
 	maintainerAddress := bdm.chain.Signing().Address()
```

Comment:
This function was removed as the BTC difficulty bot submits transactions to `LightRelayMaintainerProxy` and therefore only needs to be authorized in `LightRelayMaintainerProxy`.

However, we could in theory perform some additional checks to verify the whole setup:
- retrieve the `LightRelay` from `LightRelayMaintainerProxy` and see if `LightRelayMaintainerProxy`  is authorised in it,
- retrieve the `ReimbursementPool` from `LightRelayMaintainerProxy` and see if `LightRelayMaintainerProxy`  is authorised in it.

That would give us certainty that the whole maintainer setup is correct. If we are lacking some authorisation, we will only learn about it when attempting `retarget` which happens every 2 weeks.

Reply:
Is there a way to support both the proxied and the non-proxied mode?

If we were to support only the proxied mode, I think we should only pass the address of `LightRelayProxy` to the client and add functions to the contract allowing to expose all the `LightRelay` functions "hidden" by the proxy. This would be simpler: only one contract binding, only one address to provide.

I was not complaining about two addresses and bindings in https://github.com/keep-network/keep-core/pull/3525 because I was thinking we will support both proxied and non-proxied modes. In other words, there is some gas overhead with going through `LightRelayMaintainerProxy` so if the DAO decides to run a bot or to refund the bot runner by simple ETH transfer, the bot would work against `LightRelay` directly.

One option could be to add a flag to disable retargeting via proxy. We'd retarget via proxy by default. We would have two variations of some functions in `pkg/chain/ethereum/bitcoin_difficulty.go`: `Retarget` and `RetargetWithRefund`. Also `IsAuthorized` and `IsAuthorizedForRefund`.

What do you think?